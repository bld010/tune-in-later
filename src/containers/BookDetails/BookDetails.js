import React from 'react';
import './BookDetails.scss';

const BookDetails = ({book}) => {
  
  const {title, author, genre, description, image} = book
  
  return(
  
    <div className="BookDetails">
      <img src={image} alt={title}></img>
      <div className="bookInfo">
        <h2>{title}</h2>
        <h3>{author}</h3>
        <p>{genre} </p>
      </div>
      <div className="description">
      <p>{description}</p>
      </div>
    </div>
  )
}

export default BookDetails;