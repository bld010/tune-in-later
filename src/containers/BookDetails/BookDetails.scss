.BookDetails {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 20px;

  & .bookInfo {
  display: flex;
  height: 150px;
  flex-direction: column;
  justify-content: space-between;
  text-align: center;
  margin-left: 20px;
  

    & h2 {
      font-size: 24px;
    }

    & h3 {
      font-size: 24px;
      font-style: italic;
      color: #666888;
    }

    & h4 {
      font-size: 24px;
    }
  }

  & .description {
    width: 400px;
  }

  & img {
    height: 160px;
    width: 140px;
    border: 1px solid black;
    border-radius: 9px;
    margin: 10px 0px 0px 0px;
  }
  

  & p {
    font-size: 13px;
    margin: 0px 0px 0px 15px;
  }
}