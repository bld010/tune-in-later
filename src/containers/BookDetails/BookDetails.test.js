import React from 'react';
import { shallow } from 'enzyme';
import BookDetails from './BookDetails';

describe('BookDetails', () => {
  it('should match the snapshot with correct data passed in', () => {
    const mockBook = {
      title:'The Hell of Incessant Testing',
      author:'David Gitlen',
      genre:'Memoir',
      price: 1,
      description:'Chilling true tales from the basement of the Guaranty Bank building!',
      image:'http://www.davidgitlen.com/author.png'
    }
    const wrapper = shallow(<BookDetails 
      book={mockBook}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});