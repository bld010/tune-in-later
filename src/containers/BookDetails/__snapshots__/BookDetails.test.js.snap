// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetails should match the snapshot with correct data passed in 1`] = `
<div
  className="BookDetails"
>
  <img
    alt="The Hell of Incessant Testing"
    src="http://www.davidgitlen.com/author.png"
  />
  <div
    className="bookInfo"
  >
    <h2>
      The Hell of Incessant Testing
    </h2>
    <h3>
      David Gitlen
    </h3>
    <p>
      Memoir
       
    </p>
  </div>
  <div
    className="description"
  >
    <p>
      Chilling true tales from the basement of the Guaranty Bank building!
    </p>
  </div>
</div>
`;
